models/filtermodel.cpp

```diff
@@ -35,6 +37,40 @@ void FilterModel::setSourceModel(DocumentListModel *model)
     QSortFilterProxyModel::setSourceModel(static_cast<QAbstractItemModel*>(model));
 }
 
+int FilterModel::sortParameter() const
+{
+    return m_sortParameter;
+}
+
+void FilterModel::setSortParameter(int sortParameter)
+{
+    if (m_sortParameter == sortParameter) {
+        return;
+    }
+    
+    m_sortParameter = sortParameter;
+    Qt::SortOrder order = Qt::AscendingOrder;
+
+    switch (m_sortParameter) {
+    case Name:
+        setSortRole(DocumentListModel::Roles::FileNameRole);
+        break;
+    case Type:
+        setSortRole(DocumentListModel::Roles::FileTypeRole);
```

Comment:
This also is a bit surprising now. Choosing file type will get big groups for specific extension, but inside the file types the items are sorted, as it appears to me, oldest first. Should get secondary sorting by name.

Alternatively we could first merge the name and date sorting and add this one later.

Reply:
Explain please. You told that you want sorting by date in type sorting groups. But why is the  name sorting secondary?

Reply:
So sorting by name and sorting by date will almost never have items that are considered equal (i.e. name would need to be same for two files etc). While as for file type you will get big groups that are equal by the file type, like all the pdf files, and inside the group there is either random order or order by some secondary property. Now the secondary property seems to be date and that by showing the oldest first.

Switching the secondary order of FileType sorting to be alphabetical would make more sense. The file type part is already getting sorted alphabetically too.

Reply:
Ok. In documentlistmodel.cpp:72 `d->entries.at(index.row()).fileRead.toString(Qt::ISODate)` is added to the file extension. Will it be better to change it to file name? Or will it be better to create another role?

Reply:
Right, that's a strange thing to append there. File type should be file type and nothing more.

Actually trying removing the ISODate part I'm getting secondary ordering to be newest first, which is good enough for me.

Reply:
Ok, I'll create another role so

Reply:
Done